xlators/cluster/dht/src/dht-lock.c

```diff
@@ -907,13 +907,16 @@ dht_nonblocking_inodelk_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     char gfid[GF_UUID_BUF_SIZE] = {0};
 
     local = frame->local;
+    if (!local)
+        goto out;
```

Comment:
I think this is not necessary. The code is full of unnecessary checks.

Coverity considers this an issue because we were checking if `local` was NULL after having dereferenced it. If I'm not wrong, it's not possible to get here with a NULL `local`, so we only need to not check it.

Reply:
It looked the same to me but still, I thought of adding a check.
Anyways, I will update the code to remove that check.

Reply:
nit: could have assigned to a dht_dir_transaction_t pointer the address of local->lock[0] instead of just local, which would have also made the whole function much more readable, btw.
(or even point to local->lock[0].layout.my_layout )

Reply:
> nit: could have assigned to a dht_dir_transaction_t pointer the address of local->lock[0] instead of just local, which would have also made the whole function much more readable, btw.
> (or even point to local->lock[0].layout.my_layout )

Where exactly? I didn't get that, the only line I could see `local` is being assigned a value is at line: 905, that too it is just initialized with `NULL`.

Reply:
dht_nonblocking_inodelk_cbk()

Reply:
Untested:
```
static int32_t
dht_nonblocking_inodelk_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
                            int32_t op_ret, int32_t op_errno, dict_t *xdata)
{
    dht_local_t *local = NULL;
    int lk_index = 0, call_cnt = 0;
    char gfid[GF_UUID_BUF_SIZE] = {0};
    dht_ilock_wrap_t *my_layout;

    local = frame->local;
    lk_index = (long)cookie;

    my_layout = &(local->lock[0].layout.my_layout);

    if (op_ret == -1) {
        my_layout->op_ret = -1;
        my_layout->op_errno = op_errno;

        if (my_layout->locks[lk_index]) {
            uuid_utoa_r(my_layout->locks[lk_index]
                            ->loc.inode->gfid,
                        gfid);

            gf_msg_debug(
                this->name, op_errno,
                "inodelk failed on gfid: %s "
                "subvolume: %s",
                gfid,
                my_layout->locks[lk_index]->xl->name);
        }

        goto out;
    }

    my_layout->locks[lk_index]->locked = _gf_true;

out:
    call_cnt = dht_frame_return(frame);
    if (is_last_call(call_cnt)) {
        if (my_layout->op_ret < 0) {
            dht_inodelk_cleanup(frame);
            return 0;
        }

        dht_inodelk_done(frame);
    }

    return 0;
}
```

Reply:
Okay, that increases the code readability to a great level. Thanks :)